### ISSUE TITLE
Unable to use polymorphic serializers module

### ISSUE THREAD

**Issue by Peanuuutz** [mamoe/mirai-console#391]
When registering the serializers module in plugin config like this:
```kotlin
object TaskConfig : AutoSavePluginConfig("Task") {
    override val serializersModule = SerializersModule {
        polymorphic(Task::class) {
            subclass(ListenerTask::class)
        }
    }
}
```
The console complains about this and throws `kotlinx.serialization.SerializationException: Class 'ListenerTask' is not registered for polymorphic serialization in the scope of 'Task'. Mark the base class as 'sealed' or register the serializer explicitly.`

But in the unit test:
```kotlin
val task: Task = ListenerTask()
Json {
    serializersModule = SerializersModule {
        polymorphic(Task::class) {
            subclass(ListenerTask::class)
        }
    }
}.encodeToString(task).let(::println)
```
It actually works and prints: `{"type":"listener","id":0,"date":"2021-07-31-23-03",...}`

Here is `Task` and `ListenerTask`: (unimportant detail is omitted)
```kotlin
interface Task

@Serializable
class ListenerTask(...) : Task
```

**So is there something wrong in the YAML encoder?**

**Comment by Him188** [comment]
Yes. Yamlkt does not support polymorphic serialization at this moment.

**Comment by eav-eav-eav** [comment]
> Yes. Yamlkt does not support polymorphic serialization at this moment.

So now it's for plugin data only?

**Comment by Him188** [comment]
> > Yes. Yamlkt does not support polymorphic serialization at this moment.
> 
> So now it's for plugin data only?

Polymorphic serialization is planned but there is no exact date. Unfortunately, nested `PluginData`s are also not available yet(#146). So the only solution is to use ordinary data classes as follows:

```kotlin
object D : PluginData("name") {
    class Nested(
        val x: Int,
    )

    val nested by value(Nested(1))
}
```

**Comment by eav-eav-eav** [comment]
@Him188 Found something maybe helpful.

In [this file](https://github.com/mamoe/mirai-console/blob/6a63f1416febde02dcf14f11960e9bc6b09b2e8f/backend/mirai-console/src/internal/data/MultiFilePluginDataStorageImpl.kt), change `Yaml` on [L39](https://github.com/mamoe/mirai-console/blob/6a63f1416febde02dcf14f11960e9bc6b09b2e8f/backend/mirai-console/src/internal/data/MultiFilePluginDataStorageImpl.kt#L39) to `Yaml(yaml) { serializersModule = instance.serializersModule }`.

Others like on [L77, L78](https://github.com/mamoe/mirai-console/blob/6a63f1416febde02dcf14f11960e9bc6b09b2e8f/backend/mirai-console/src/internal/data/MultiFilePluginDataStorageImpl.kt#L77-L78), [L87](https://github.com/mamoe/mirai-console/blob/6a63f1416febde02dcf14f11960e9bc6b09b2e8f/backend/mirai-console/src/internal/data/MultiFilePluginDataStorageImpl.kt#L87) could apply this change as well.

It works as expected in this form:
```yaml
type: listener
value:
  id: 0
  date: ...
```

**Comment by Him188** [comment]
@Peanuuutz Thank you so much for the investigation! I will schedule a compability review then hopefully release the feature by 2.8.0 or 2.9.0.
